File: StackSearch/StackSearch/Models/APIParams.swift
//
//  APIParams.swift
//  StackSearch
//
//  Created by Saiful I. Wicaksana on 12/08/18.
//  Copyright © 2018 icaksama. All rights reserved.
//

import Foundation

internal class APIParams {
    
    fileprivate let defaults = UserDefaults.standard
    internal var pagesize: Int = 10
    internal var fromdate: Int = 0
    internal var todate: Int = 0
    internal var order: String = "desc"
    internal var sort: String = "activity"
    internal var tagged: String = ""
    internal var intitle: String = "ios"
    
    init() {
        self.pagesize = defaults.integer(forKey: "pagesize") == 0 ? 10 : defaults.integer(forKey: "pagesize")
        self.fromdate = defaults.integer(forKey: "fromdate")
        self.todate = defaults.integer(forKey: "todate")
        self.order = defaults.string(forKey: "order") == nil ? "desc" : defaults.string(forKey: "order")!
        self.sort = defaults.string(forKey: "sort") == nil ? "activity" : defaults.string(forKey: "sort")!
        self.tagged = defaults.string(forKey: "tagged") == nil ? "" : defaults.string(forKey: "tagged")!
        self.intitle = defaults.string(forKey: "intitle") == nil ? "ios" : defaults.string(forKey: "intitle")!
    }
    
    init(pagesize: Int, fromdate: Int, todate: Int, order: String, sort: String, tagged: String = "", intitle: String = "") {
        self.pagesize = pagesize
        defaults.set(pagesize, forKey: "pagesize")
        self.fromdate = fromdate
        defaults.set(fromdate, forKey: "fromdate")
        self.todate = todate
        defaults.set(todate, forKey: "todate")
        self.order = order
        defaults.set(order, forKey: "order")
        self.sort = sort
        defaults.set(sort, forKey: "sort")
        self.tagged = tagged
        defaults.set(tagged, forKey: "tagged")
        self.intitle = intitle
        defaults.set(intitle, forKey: "intitle")
    }
}

File: AStackSearch/app/src/main/java/com/icaksama/astacksearch/models/APIParams.java
package com.icaksama.astacksearch.models;

import android.content.Context;

import com.icaksama.astacksearch.utilities.SharedPreferenceHelper;

/**
 * Created by icaksama on 13/08/18.
 */

public class APIParams {

    private SharedPreferenceHelper mPrefs = null;
    public long pagesize = 10;
    public long fromdate = 0;
    public long todate = 0;
    public String order = "desc";
    public String sort = "activity";
    public String tagged = "";
    public String intitle = "ios";

    public APIParams(Context context) {
        mPrefs = new SharedPreferenceHelper(context);
        this.pagesize = mPrefs.getLong("pagesize") == 0 ? 10 : mPrefs.getLong("pagesize");
        this.fromdate = mPrefs.getLong("fromdate");
        this.todate = mPrefs.getLong("todate");
        this.order = mPrefs.getString("order").trim().isEmpty() ? "desc" : mPrefs.getString("order");
        this.sort = mPrefs.getString("sort").trim().isEmpty() ? "activity" : mPrefs.getString("sort");
        this.tagged = mPrefs.getString("tagged").trim().isEmpty() ? "" : mPrefs.getString("tagged");
        this.intitle = mPrefs.getString("intitle").trim().isEmpty() ? "android" : mPrefs.getString("intitle");
    }

    public APIParams(Context context, long pagesize, long fromdate, long todate, String order, String sort, String tagged, String intitle) {
        mPrefs = new SharedPreferenceHelper(context);
        this.pagesize = pagesize;
        mPrefs.setLong("pagesize", pagesize);
        this.fromdate = fromdate;
        mPrefs.setLong("fromdate", fromdate);
        this.todate = todate;
        mPrefs.setLong("todate",todate);
        this.order = order;
        mPrefs.setString("order", order);
        this.sort = sort;
        mPrefs.setString("sort", sort);
        this.tagged = tagged;
        mPrefs.setString("tagged", tagged);
        this.intitle = intitle;
        mPrefs.setString("intitle", intitle);
    }
}
